lib/core/src/sync/mod.rs

```diff
@@ -90,13 +97,26 @@ impl SyncService {
                 return;
             }
 
-            let mut sync_trigger = self.sync_trigger.lock().await;
+            let mut local_sync_trigger = self.sync_trigger.lock().await;
+            let mut remote_sync_trigger = match self.new_listener().await {
+                Ok(trigger) => trigger,
+                Err(err) => {
+                    log::warn!("Could not start remote sync trigger: {err:?}");
+                    return;
+                }
+            };
             let mut event_loop_interval = tokio::time::interval(Duration::from_secs(30));
 
+            log::debug!("Starting real-time sync event loop");
+
             loop {
                 tokio::select! {
                     _ = event_loop_interval.tick() => self.run_event_loop().await,
-                    Some(_) = sync_trigger.recv() => {
+                    Some(_) = local_sync_trigger.recv() => {
+                        self.run_event_loop().await;
+                        event_loop_interval.reset();
+                    },
+                    Some(Ok(_)) = remote_sync_trigger.next() => {
```

Comment:
Did you check the case where the internet connection is lost? I guess ere it returns an error. Not sure if we should recreate the listener stream or the old one recovers automatically. In breez-sdk-greenlight I think we re-create the streams in that case.